tasks/akeyless/src/main/java/com/walmartlabs/concord/plugins/akeyless/model/TaskParamsImpl.java

```diff
@@ -138,6 +143,49 @@ public Action action() {
         return action(input);
     }
 
+    /**
+     * @param o Input value with may be a String or Map of Concord secret
+     *          info (org, name, password)
+     * @param secretExporter for access Concord's Secrets API
+     * @return String value from direct input or exported Secret value
+     */
+    @SuppressWarnings("unchecked")
+    private static String stringOrSecret(Object o, SecretExporter secretExporter) {
+        if (o instanceof String) {
+            return (String) o;
+        }
+
+        if (! (o instanceof Map)) {
+            throw new IllegalArgumentException("Invalid data type given for sensitive argument. Must be string or map.");
+        }
+
+        ((Map<?, ?>) o).forEach((key, value) -> {
+            if (!(key instanceof String)) {
+                throw new IllegalArgumentException("Non-string key used for secret definition");
+            }
+
+            if (!(value instanceof String)) {
+                throw new IllegalArgumentException("Non-string value used for secret definition");
+            }
+        });
+
+        return exportSecret((Map<String, String>) o, secretExporter);
+    }
+
+    private static String exportSecret(Map<String, String> secretInfo, SecretExporter secretExporter) {
+        final String o = secretInfo.get("org");
+        final String n = secretInfo.get("name");
+        final String p = secretInfo.getOrDefault("password", null);
+
+        return secretCache.get(o + n, () -> {
```

Comment:
with `o + n` as key we can get collisions:
```
o = dev
n = toolsKey
---
o = devtools
n = Key
```
looks like `o + '/' + n` is better.

Reply:
and I think it's better to hide how cache handle keys...
something like this:
```
secretCache.get(o, n, () -> ...);
```
and in SecretCache:
```
...
String get(String org, String name, Supplier<String> lookup) {
    String key = buildKey(org, name);
    ...
}

private static String buildKey(String org, String name) {
   return org + "/" + name;
}
...
```